csrc/device_lower/validation.cpp

```diff
@@ -663,6 +663,11 @@ class ExprValidator : public OptOutDispatch {
     // https://docs.nvidia.com/cutlass/media/docs/cpp/blackwell_functionality.html#scale-factor-layouts
     if (block_scaling_factor->hasAllocation()) {
       isValidBlockScaleSwizzle(block_scaling_factor);
+      NVF_ERROR_EQ(
+          bqop->isSwizzledScales(),
+          true,
+          "Block scaling factor with allocation domain requires swizzled "
+          "scales.");
```

Comment:
note: for multi-gpu, where allocation domain would be used for sharding, we might run into issues here. But we can fix that at a later point.